dataservice.js

```diff
@@ -0,0 +1,47 @@
+
+
+const combinedProductMap = new Map();
+const productMap = new Map();
+const categoryMap = new Map();
+
+// function to load products JSON and convert it to a map.
+let getProducts = () => {
+    const productsJson = require('./my-app-react/src/data/products.json');
+    if (productMap.size === 0) {
+        for (let product of productsJson.products) {
```

Comment:
Can't this be const?
```suggestion
        for (const product of productsJson.products) {
```